x-pack/docs/en/security/securing-communications/update-tls-certificates.asciidoc

```diff
@@ -0,0 +1,784 @@
+
+[[use-rolling-restarts]]
+.Rolling restarts are preferred
+****
+While it's possible to do an in-place update for security certificates, using
+a <<restart-cluster-rolling,rolling restart>> on your cluster is safer. An in-place update avoids some
+complications of a rolling restart, but incurs the following risks:
+
+* If you use PEM files, your certificate and key are in separate files. You
+must update both files _simultaneously_ or the node might experience a temporary
+period where it cannot establish new connections.
+* Updating the certificate and key does not automatically force existing
+connections to refresh. This means that even if you make a mistake, a node can
+seem like it's functioning but only because it still has existing connections.
+It's possible that a node will be unable to connect with other nodes, rendering
+it unable to recover from a network outage or node restart.
+****
+
+[[update-node-certs-same]]
+=== Update security certificates with the same CA
+++++
+<titleabbrev>With the same CA</titleabbrev>
+++++
+
+This procedure assumes that the you have access to the CA certificate and key
+that was originally generated (or otherwise held by your organization) and used
+to sign the certificate that is currently in use. It also assumes that the
+clients connecting to {es} on the HTTP layer are configured to trust the CA
+certificate.
+
+If you have access to the CA used to sign your existing certificates, you only
+need to replace the certificates and keys for each node in your cluster. If you
+replace your existing certificates and keys on each node use the same filenames,
+{es} reloads the files starts using the new certificates and keys.
+
+You don't have to restart each node, but doing so forces new TLS connections and is <<use-rolling-restarts,a recommended practice>> when updating certificates.
+Therefore, the following steps include a node restart after updating each
+certificate.
+
+The following steps provide instructions for generating new node certificates
+and keys for both the transport layer and the HTTP layer. You might only need
+to replace one of these layer's certificates depending on which of your
+certificates are expiring.
+
+[[cert-password-updates]] 
+IMPORTANT: If your keystore is password protected, the password
+is stored in the {es} secure settings, _and_ the password needs to change, then
+you must perform a <<restart-cluster-rolling,rolling restart>> on your cluster.
+You must also use a different file name for the keystore so that {es} doesn't
+reload the file before the node is restarted.
+
+TIP: If your CA has changed, complete the steps in 
+<<update-node-certs-different,update security certificates with a different CA>>.
+
+[[node-certs-same-transport]]
+==== Generate a new certificate for the transport layer
+The following examples use PKCS#12 files, but the same steps apply to JKS
+keystores.
+
+Rather than run the `elasticsearch-certutil` tool once for each node, create an
```

Comment:
This assume that each node uses a different keystore for the transport layer. Our [existing docs suggestion](https://www.elastic.co/guide/en/elasticsearch/reference/current/security-basic-setup.html) and the default in 8.0.0 will be to use the same keystore for all nodes. Should we tackle this case here too for consistency, instead of the one we currently do ? Or at least point to it ?

Reply:
True, we use the same keystore for all nodes in our docs for configuring security. Do we anticipate that users will follow that convention? Furthermore, do we _recommend_ using a single keystore rather than generating a keystore for each node? When updating certificates with a different CA, do we also recommend using the same keystore for every node? If so, then we should parallel those procedures, unless there's a compelling reason not to. From my understanding, it's safer to have a keystore for each node so that if a keystore is compromised, users only need to update a single keystore rather than on every node when using the same one throughout the cluster.

Reply:
> From my understanding, it's safer to have a keystore for each node

It is, but it's also more complicated and we've attempted to strike a balance. I think we need to sync our suggestions or at least include docs here for the case that we cover in the configuration docs too

Reply:
I'm ok with having these instructions parallel the initial configuration docs so that we're using one keystore for all nodes. I chatted with @BobBlank12 recently, and he mentioned using `instances.yml` to generate a keystore for each node. I just wanted to ensure that we're using the method in the docs that we intend users to employ in the real world.